features/karaf/src/main/feature/feature.xml

```diff
@@ -244,6 +244,7 @@
   <feature name="esh-storage-mapdb" version="${project.version}">
     <feature>esh-base</feature>
     <bundle>mvn:org.eclipse.smarthome.storage/org.eclipse.smarthome.storage.mapdb/${project.version}</bundle>
+    <bundle dependency="true">mvn:org.mapdb/mapdb/1.0.9</bundle>
```

Comment:
dependency means that this is installed if some dependency (package-import) is not satisfied.
So, it will be installed, because it is used by storage.mapdb.

If another bundle depends on mapdb 1.0.10 some time, and this version also satisfy the dependency of storage.mapdb, there is only one (the 1.0.10) installed in the runtime.